src/hosting-json.node.js

```diff
@@ -28,26 +31,28 @@ async function streamToString(stream) {
  */
 async function getGzippedFileAsJson(jsonPath) {
   const readStream = fs.createReadStream(jsonPath);
-  const text = await streamToString(readStream);
+  const text = await streamToBuffer(readStream);
   const unzipped = await gunzip(text);
   return unzipped.toString();
 }
 
 /**
  * Loads JSON data from a file path.
  * @param {string} jsonPath - The path to the JSON file.
- * @returns {Promise<object>} A promise that resolves to the parsed JSON object.
+ * @returns {Promise<string[]>} A promise that resolves to the parsed JSON object.
  */
 async function loadJSON(jsonPath) {
   const jsonBuffer = jsonPath.endsWith(".gz")
     ? await getGzippedFileAsJson(jsonPath)
     : await readFile(jsonPath);
-  return JSON.parse(jsonBuffer);
+  // TODO (simon) should we check that the parsed JSON is a list of strings?
```

Comment:
question: typescript itself can't guarantee that what comes out of `JSON.parse` is actually a list of strings. would it be worth adding some runtime checking to validate?